targets/TARGET_NORDIC/TARGET_NRF5x/qspi_api.c

```diff
@@ -247,6 +251,11 @@ qspi_status_t qspi_frequency(qspi_t *obj, int hz)
 
 qspi_status_t qspi_write(qspi_t *obj, const qspi_command_t *command, const void *data, size_t *length)
 {
+    // length needs to be rounded up to the next WORD (4 bytes)
+    if ((*length & WORD_MASK) > 0) {
```

Comment:
Why does the transfer size need to be 4 byte aligned? I don't see this as a requirement in the documentation `qspi_api.h`.

Reply:
Turns out this is a limitation of the NRF52. Thanks for updating the documentation @j3hill

Reply:
Yes. This is a Nordic requirement. In the nRF52840 product specification document (https://infocenter.nordicsemi.com/pdf/nRF52840_OPS_v0.5.pdf ) it states that for qspi read and write operations: "The length must be a multiple of 4 bytes." To help make this clear going forward, I've updated the length comments in "qspi_api.h" to note that support for non 4-byte (WORD) aligned reads/writes is implementation defined.

Reply:
@j3hill - I do not understand the comment: "support for non 4-byte (WORD) aligned reads/writes is implementation defined". 4-bytes alignment is Nordic limitation. Other targets might have other limitations.
Currently qspi_hal has no limitation on read/write granularity - so minimal size is 1byte. 
The way I understand it the remark should be something like: "support for any specific target alignment requirement is - target implementation defined"

Reply:
I dont not understand neither rereading this thread. Similar limitations we have in flashing. How do we cover it there? via `get_program_page_size()` ?

Reply:
@donatieng  , @0xc0170 - if you're referring to FalshIAP, it is covered by a **hal API _flash_get_page_size()_** which is implemented in each target, and surface its requirement to the driver FlashIAP. The FlashIAP driver then handles the program address alignment and granularity requirements with its own internal buffer.

Driver:
*******
_uint32_t FlashIAP::get_page_size() const
{
    return flash_get_page_size(&_flash);
}_

HAL:
*****
_/** Get page size
 *
 * The page size defines the writable page size
 * @param obj The flash object
 * @return The size of a page
 */
uint32_t flash_get_page_size(const flash_t *obj);_

Reply:
@offirko you are referring to the internal flash. The external QSPI flash blockdevice is defined here: https://github.com/ARMmbed/mbed-os/blob/master/components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp#L468-L472

Reply:
@marcuschangarm - I know.

I was simply showing @0xc0170 and @donatieng that in FlashIAP the **driver** handles alignment and write-granularity using its own internal buffer, and based on page_size upstreamed from **hal** .
My original suggestion was to do the same for QSPI